pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -156,6 +159,140 @@ func decodeIngressRuleName(name string) (ingressName string, ruleNum, pathNum in
 	return
 }
 
+// ConvertIngressV1alpha3 converts from ingress spec to Istio Gateway + VirtualServices
+func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model.Config, model.Config) {
+
+	gateway := &networking.Gateway{
+		Selector: model.IstioIngressWorkloadLabels,
+	}
+
+	for _, tls := range ingress.Spec.TLS {
+		gateway.Servers = append(gateway.Servers, &networking.Server{
+			Port: &networking.Port{
+				Number:   443,
+				Protocol: string(model.ProtocolHTTP),
+				Name:     "http-ingress-443",
+			},
+			Hosts: tls.Hosts,
+			Tls: &networking.Server_TLSOptions{
+				HttpsRedirect:     false,
+				Mode:              networking.Server_TLSOptions_SIMPLE,
+				ServerCertificate: path.Join(model.IngressCertsPath, ingress.Namespace, tls.SecretName, model.IngressCertFilename),
+				CaCertificates:    path.Join(model.IngressCertsPath, ingress.Namespace, tls.SecretName, model.IngressKeyFilename),
+			},
+		})
+	}
+
+	gateway.Servers = append(gateway.Servers, &networking.Server{
+		Port: &networking.Port{
+			Number:   80,
+			Protocol: string(model.ProtocolHTTP),
+			Name:     "http-ingress-80",
+		},
+	})
+
+	virtualService := &networking.VirtualService{
+		Hosts:    []string{"*"},
+		Gateways: []string{model.IstioIngressGatewayName},
+	}
+
+	var httpRoutes []*networking.HTTPRoute
+	for _, rule := range ingress.Spec.Rules {
+		if rule.HTTP == nil {
+			log.Warnf("invalid ingress rule for host %q, no paths defined", rule.Host)
```

Comment:
`Info`. There's nothing really wrong with this (users freak out about `warnings` in logs, we see them posted all the time when they mean nothing; I want as few warnings logged as possible, and only for really bad stuff).